src/main/java/com/palantir/ri/ResourceIdentifier.java

```diff
@@ -328,7 +332,7 @@ public static ResourceIdentifier valueOf(String rid) {
     public static ResourceIdentifier of(String rid) {
         ResourceIdentifier resultRid = tryOf(rid);
         if (resultRid == null) {
-            throw new IllegalArgumentException("Illegal resource identifier format: " + rid);
+            throw new SafeIllegalArgumentException("Illegal resource identifier format", UnsafeArg.of("rid", rid));
```

Comment:
nit: This isn't incorrect now, but shall we also update the javadoc to `@throws SafeIllegalArgumentException`?

Reply:
In general we don’t really want folks to catch the safe variants, so we don’t want folks to rely on the safe variant being api, but that would be more specific in javadoc. I could go either way :-)

Reply:
Ok. If people shouldn't be catching _only_ the safe variants, then having the API specify the superclass lets them know what to catch instead.

No strong stance either, but leaning towards keeping as-is with the principle "the @throws block in the javadoc should specify what devs to should catch".

Reply:
Going to merge given this was the only comment.